### PR TITLE
CB-11762 Adding listUsers and listGroups to thunderhead-mock

### PR DESCRIPTION
**Pull request by tiborm**
See detailed description in the commit message.

### CODE REVIEW

**Review by akanto — APPROVED** [review]

**Review by lajosrodek — APPROVED** [review]
LGTM

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+@RestController
+public class MockThunderheadIAMController {
+
+    @CrossOrigin(origins = "*")
+    @PostMapping("/iam/getUser")
+    public HashMap<String, String> getUser(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
```
Comment (lajosrodek) [inline mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java:17]: Not a big deal, perhaps using the interface `Map` for the return type would be better.

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return new HashMap<>();
+    }
+
+    @CrossOrigin(origins = "*")
+    @PostMapping("/iam/listGroups")
+    public ArrayList<String> listGroups(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
```
Comment (lajosrodek) [inline mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java:23]: Like above, using `List` would perhaps make more sense. But this is a mock implementation (hopefully a short-lived one), so I have no strong feelings. 😃

**Review by lajosrodek — APPROVED** [review]
LGTM. Had 2 small comments, feel free to ignore them as this is a quick-and-dirty mock impl.

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java (hunk 2 above)
Comment (akanto) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java:23]: Unfortunately, checkstyle has a strong feeling against HashMap and ArrayList :)

Comment (tiborm) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java:23]: Yeah, am on it.

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java (hunk 1 above)
Comment (tiborm) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java]: thx, done

File: mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java (hunk 2 above)
Comment (tiborm) [reply mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/controller/MockThunderheadIAMController.java]: thx, done

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Pull request merged**